github: update PR template to include a checklist (#14676)
snapcore/snapd: .github/PULL_REQUEST_TEMPLATE.md

Thanks for helping us make a better snapd!

Have you signed the [license
agreement](https://www.ubuntu.com/legal/contributors) and read the [contribution
guide](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md)?

> Please make sure that the following items have been verified: (please leave a
> note if one of those is not applicable):

Reviewer: ```suggestion
Please check the following items and add a comment for any that may not apply.
```

> - [ ] there are unit tests covering your change

Reviewer: ```suggestion
- [ ] unit tests are included for the proposed change
```

- [ ] unit tests are passing (./run-checks --unit)
- [ ] static files tests are passing (./run-checks --static)

> - [ ] snapd snap build works with latest snapcraft (snapcraft --verbose)

Reviewer: ```suggestion
- [ ] the latest snapcraft version (snapcraft --verbose) builds the snapd snap
```

> - [ ] snapd snap build works with latest snapcraft (snapcraft --verbose)

Reviewer 2: This is (1) rather time consuming and (2) strictly speaking incomplete as it does not include deb artifices.

There is a balance between  running tests upfront to save time/noise in CI failure VS not duplicating expensive test.

> - [ ] the change needs an entry in the changelog

Reviewer: ```suggestion
- [ ] changelog needs to reference this change
```

> - [ ] the change needs an entry in the NEWS.md file

Reviewer 2: During the last few iterations I concluded this is not useful.

(1) We need 100% certainty that nothing is missed, which means I need do what I do and run script to detect all PR's between previous ad current release branch, minus cherry-picked items. From this, test only development as well as internal changes are removed. Then related items (e.g. interface changes) are grouped. Then PR's related to same topic (e.g. components) are summarised. I think this is not possible to get right with run-time news updates.

(2) In addition, having entries in NEWS.md for upcoming release means merging back from release branch is not super trivial as it is without.
